### ISSUE TITLE
label-has-associated-control not checking an actual valid configuration

### ISSUE THREAD

**Issue by tomdglenn91** [jsx-eslint/eslint-plugin-jsx-a11y#956]
Apologies if I've misunderstood the rule or configured this wrong.

I want to ensure that every (custom) input element has an associated label, and it's configured correctly. So the following works out the box:

Given the following configuration:
```json
    'jsx-a11y/label-has-associated-control': [
      'warn',
      {
        labelComponents: ['FormLabel'],
        controlComponents: ['Input', 'Select']
      },
    ],
```

I expect this to fail, as no HtmlFor, which it fails.
```tsx
<>
   <FormLabel>Foo</FormLabel>
   <Input name="foo" />
</>
```

I expect this to fail, as the htmlFor doesn't link to anything, but it doesn't fail
```tsx
<>
   <FormLabel htmlFor="foo">Foo</FormLabel>
   <Input name="foo" />
</>
```

Then expect this to pass
```tsx
<>
   <FormLabel htmlFor="foo">Foo</FormLabel>
   <Input id="foo" name="foo" />
</>
```

Given that:
> This rule checks that any label tag (or an indicated custom component that will output a label tag) either (1) wraps an input element (or an indicated custom component that will output an input tag) or (2) has an htmlFor attribute and that the label tag has text content.

I feel like my expected behaviour is not the reality, but I wasn't quite sure. 

Should my expectation work, or am I just misunderstanding the library/rule? Thanks.

**Comment by ljharb** [comment]
I would agree that `has an htmlFor attribute and that the label tag has text content.` matches the reality and not your expectation.

However, I would expect that 2 should also be checking that the `htmlFor` value matches the label's `id` value.

That seems like a useful option to add.

**Comment by michaelfaith** [comment]
I have this change ready to go, but want to land this change making the error messages more specific to the assert type first: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/pull/1041

The follow-up change adding the option proposed here builds on that change.

**Comment by michaelfaith** [comment]
This should be ready for review, when you're able: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/pull/1042
